craft_application/services/lifecycle.py

```diff
@@ -331,6 +332,23 @@ def _get_local_keys_path(self) -> Path | None:
         """
         return None
 
+    def save_pack_state(self, packages: list[str]) -> None:
```

Comment:
Based on the conversation today, what if we had a `StateService` that would manage the state and communicate it between the managed instance and the outer instance?

Reply:
`StateService` sounds like a good solution for this case, but I think we should add it in a separate PR to avoid bringing too much complexity to this change.